src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -137,7 +137,7 @@ final case class PurposeApiServiceImpl(
       _              <- assertOrganizationIsAConsumer(organizationId, seed.consumerId)
       eServiceUUID   <- eServiceId.toFutureUUID
       eService       <- catalogManagementService.getEServiceById(eServiceUUID)
-      _ <- if (eService.mode == Deliver) Future.failed(EServiceNotInReceiveMode(eService.id)) else Future.unit
+      _ <- if (eService.mode == Receive) Future.failed(EServiceNotInDeliverMode(eService.id)) else Future.unit
```

Comment:
this was fine 😄 

As a recap:
- in the `createPurposeFromEService` the eservice should be in mode `Receive`
- in the `updatePurpose` the eservice should be in mode `Deliver`

@beetlecrunch do you confirm?

Btw I suggest to test the success condition and fail otherwise, as an example 
```
if (eService.mode == Receive) Future.unit else Future.failed(EServiceNotInDeliverMode(eService.id))
```
this way if we'll add more values to the enum, we can still be on the safe authorization side